src/main/java/io/vertx/core/shareddata/AsyncMapStream.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2011-2017 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.shareddata;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Closeable;
+import io.vertx.core.Handler;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * An {@link AsyncMap} keys, values, or entries {@link ReadStream stream}.
+ * <p>
+ * Make sure to invoke {@link #close(Handler)} after usage to avoid leaking resources.
```

Comment:
should `close()` be used when the stream is completed ?

Reply:
Yes, either ended or failed.

Reply:
I think that close should not have to be used when the stream is ended or failed. It should rather be used when the user wants to dispose the stream prematurely (for instance there is an error during the iteration and the developer want to release the sream resources).

Reply:
I'm not sure I understand. Do you mean close should be automatically invoked if the stream fails (before `exceptionHandler` is called) or ends (before `endHandler` is called)? And the user should have to call it only if he wants to end things prematurely (because he doesn't need the rest of the data or an error happened downstream)?

Reply:
yes

Reply:
Done, including cluster managers updates